xarray/conventions.py

```diff
@@ -235,6 +235,11 @@ def encode_cf_variable(var, needs_copy=True, name=None):
     var = maybe_default_fill_value(var)
     var = maybe_encode_bools(var)
     var = ensure_dtype_not_object(var, name=name)
+
+    if var.encoding.get('grid_mapping', None) is not None:
```

Comment:
There's no need to supply a default value of `None` with `.get()`, that is already the default:
```suggestion
    if var.encoding.get('grid_mapping') is not None:
```

Reply:
But actually: are there any cases where someone would explicitly have `grid_mapping=None` in `encoding`? If not, then let's just check:
```suggestion
    if 'grid_mapping' in var.encoding:
```

Reply:
Probably not.  That detail should go in documentation somewhere.  Where would you suggest?

Reply:
Code change is done.  Docs change is not.  What page should that go on? IO/NetCDF?